core/server/worker/src/main/java/alluxio/worker/block/meta/StorageDir.java

```diff
@@ -57,16 +57,23 @@
   private Map<Long, Set<Long>> mSessionIdToTempBlockIdsMap;
   private AtomicLong mAvailableBytes;
   private AtomicLong mCommittedBytes;
+  private AtomicLong mReservedBytes;
+  private AtomicLong mUsedReservedBytes;
   private String mDirPath;
   private int mDirIndex;
   private StorageTier mTier;
 
-  private StorageDir(StorageTier tier, int dirIndex, long capacityBytes, String dirPath,
-      String dirMedium) {
+  private StorageDir(StorageTier tier, int dirIndex, long capacityBytes, long reservedBytes,
+      String dirPath, String dirMedium) {
     mTier = Preconditions.checkNotNull(tier, "tier");
     mDirIndex = dirIndex;
     mCapacityBytes = capacityBytes;
-    mAvailableBytes = new AtomicLong(capacityBytes);
+    if (capacityBytes < reservedBytes) {
+      reservedBytes = capacityBytes / 20;
```

Comment:
Do we need to make sure it is not too small either?

Reply:
I feel like that would require another prop to get a test pass.